configure: do not force a static link for non-installed programs

Otherwise, it tries to link against a static libz, which may not exist
in a shared-only system.

Signed-off-by: "Yann E. MORIN" <yann.morin.1998@free.fr>

diff -durN mysql-5.1.73.orig/configure.in mysql-5.1.73/configure.in
--- mysql-5.1.73.orig/configure.in	2014-12-22 00:04:46.550508208 +0100
+++ mysql-5.1.73/configure.in	2014-12-22 00:05:56.415307480 +0100
@@ -562,7 +562,6 @@
   AC_MSG_ERROR([MySQL requires an ANSI C compiler (and a C++ compiler). Try gcc. See the Installation chapter in the Reference Manual.])
 fi
 
-NOINST_LDFLAGS="-static"
 
 static_nss=""
 STATIC_NSS_FLAGS=""
